bycycle/group/features.py

```diff
@@ -74,91 +91,154 @@ def compute_features_2d(sigs, fs, f_range, compute_features_kwargs=None,
 
-    # Drop `return_samples` argument, as this is set directly in the function call
-    compute_features_kwargs = [compute_features_kwargs] if \
-        isinstance(compute_features_kwargs, dict) else compute_features_kwargs
+    kwargs = {} if kwargs is None else kwargs
+    kwargs = [kwargs] if isinstance(kwargs, dict) else list(kwargs)
 
-    [kwargs.pop('return_samples', None) for kwargs in compute_features_kwargs]
+    # Drop return_samples argument, as it is set directly in the function call
+    for kwarg in kwargs:
+        kwarg.pop('return_samples', None)
 
     n_jobs = cpu_count() if n_jobs == -1 else n_jobs
 
-    with Pool(processes=n_jobs) as pool:
+    if axis == 0:
+        # Compute each signal's independently and in paralllel
+        with Pool(processes=n_jobs) as pool:
 
-        if len(compute_features_kwargs) > 1:
-            # Map iterable sigs and kwargs together
-            mapping = pool.imap(partial(_proxy, fs=fs, f_range=f_range,
-                                        return_samples=return_samples),
-                                zip(sigs, compute_features_kwargs))
+            if len(kwargs) > 1:
+                # Map iterable sigs and kwargs together
+                mapping = pool.imap(partial(_proxy_2d, fs=fs, f_range=f_range,
+                                            return_samples=return_samples),
+                                    zip(sigs, kwargs))
 
-        else:
-            # Only map sigs, kwargs are the same for each mapping
-            mapping = pool.imap(partial(compute_features, fs=fs, f_range=f_range,
-                                        return_samples=return_samples,
-                                        **compute_features_kwargs[0]),
-                                sigs)
+            else:
+                # Only map sigs, kwargs are the same for each mapping
+                mapping = pool.imap(partial(compute_features, fs=fs, f_range=f_range,
+                                            return_samples=return_samples,
+                                            **kwargs[0]),
+                                    sigs)
 
-        if return_samples is True:
-            df_features, df_samples = zip(*progress_bar(mapping, progress, len(sigs)))
+            df_features = list(progress_bar(mapping, progress, len(sigs)))
 
-            df_features = list(df_features)
-            df_samples = list(df_samples)
+    elif axis is None:
+        # Compute features after flattening the 2d array (i.e. calculated across a 1d signal)
+        sig_flat = sigs.flatten()
 
-        else:
-            df_features = list(progress_bar(mapping, progress, len(sigs)))
+        center_extrema = kwargs[0].pop('center_extrema', 'peak')
+
+        df_flat = compute_features(sig_flat, fs=fs, f_range=f_range, return_samples=True,
+                                   center_extrema=center_extrema, **kwargs[0])
+
+        # Reshape the dataframe into original sigs shape
```

Comment:
so, this feels a bit like it's an independent, and potentially useful task: to 'epoch' a dataframe of measured values. 

Could this be a sub-function? As well as increasing modularity, I feel like it could be useful be itself. For example, if I've already computed features on a long continuous signal, I can imagine wanting to split it up into time sections. 

Is there a relatively easy way to turn this code section into something like:
`epoch_df(df, epoch_lengths)`
which would take a single df, and split it up into a list of dfs, each representing `epoch_length` samples?
This function could be used here, and offered as a general utility, I think?

Reply:
I refactored this to a new func  in `utils.dataframes`.